smart-contracts/contracts/cl-vault/src/rewards/distribution.rs

```diff
@@ -87,80 +116,119 @@ pub fn handle_collect_spread_rewards_reply(
     let mut rewards = CURRENT_REWARDS.load(deps.storage)?;
     rewards.update_rewards(response.collected_spread_rewards)?;
 
-    // update the rewards map against all user's locked up vault shares
-    distribute_rewards(deps, rewards)?;
+
+    // calculate and update the strategist fee
+    let vault_config = VAULT_CONFIG.load(deps.storage)?;
+    let strategist_fee = rewards.sub_ratio(vault_config.performance_fee)?;
+    STRATEGIST_REWARDS.update(deps.storage, |old| old.add(strategist_fee))?;
```

Comment:
Are we double counting Strategist spread rewards? See function handle_collect_spread_rewards_reply

Reply:
Yes it was the issue. Calculation is fine now